src/SDK/Common/Configuration/Resolver/SdkConfigurationResolver.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Common\Configuration\Resolver;
+
+use Nevay\SPI\ServiceLoader;
+use Nevay\SPI\ServiceProviderDependency\PackageDependency;
+use OpenTelemetry\Config\SDK\Configuration\Environment\EnvSourceProvider;
+use OpenTelemetry\Config\SDK\Configuration\Environment\EnvSourceReader;
+use OpenTelemetry\Config\SDK\Configuration\Environment\LazyEnvSource;
+use OpenTelemetry\Config\SDK\Configuration\Environment\PhpIniEnvSource;
+use OpenTelemetry\Config\SDK\Configuration\Environment\ServerEnvSource;
+use OpenTelemetry\SDK\Common\Configuration\Configuration;
+
+/**
+ * @internal
+ */
+#[PackageDependency('open-telemetry/sdk-configuration', '*')]
+class SdkConfigurationResolver implements ResolverInterface
+{
+    private readonly EnvSourceReader $reader;
+
+    public function __construct()
+    {
+        $envSources = [];
+        $envSources[] = new ServerEnvSource();
+        $envSources[] = new PhpIniEnvSource();
+
+        /** @var EnvSourceProvider $envSourceProvider */
+        foreach (ServiceLoader::load(EnvSourceProvider::class) as $envSourceProvider) {
+            $envSources[] = new LazyEnvSource($envSourceProvider->getEnvSource(...));
```

Comment:
so the order here is $_SERVER, php.ini, then any available dotenv implementation (and lazy so that they only process if a variable is requested and not found earlier)?